src/libcore/slice.rs

```diff
@@ -684,13 +666,11 @@ macro_rules! iterator {
             fn next(&mut self) -> Option<$elem> {
                 // could be implemented with slices, but this avoids bounds checks
                 unsafe {
-                    ::intrinsics::assume(!self.ptr.is_null());
-                    ::intrinsics::assume(!self.end.is_null());
```

Comment:
I'm not sure why these assumes were added in the first place, they showed up in f9ef8cd55512842f2481aac6332dbfb92df58c52 without anything relevant in the commit message, but they aren't valid.